packages/server/tests/helpers/utils.ts

```diff
@@ -372,4 +383,229 @@ export class Utils {
+            console.info('Writing heap snapshot...');
+            await Utils.writeHeapSnapshotAsync();
+          }
+        }
+      } catch (error) {
+        console.error('Error capturing memory leaks:', error);
+      }
+    });
+  }
+
+  /**
+   * Generates a comment indicating a memory leak detected during tests.
+   * @param {string} testTitle The title of the current test.
+   * @param {HeapDifferenceStatistics} statsDiff The difference in memory statistics indicating the leak.
+   * @returns {string} The formatted comment.
+   */
+  private static generateMemoryLeakComment(testTitle: string, statsDiff: HeapDifferenceStatistics): string {
+    const commentHeader = '## 🚨 Memory Leak Detected 🚨';
+    const summary = `A potential memory leak has been detected in the test titled \`${testTitle}\`. This may impact the application's performance and stability.`;
+    const detailsHeader = '### Details';
+    const formattedStatsDiff = this.formatHeapDifferenceStatistics(statsDiff);
+    const recommendationsHeader = '### Recommendations';
+    const recommendations =
+      'Please investigate the memory allocations in this test, focusing on objects that are not being properly deallocated.';
+
+    return `${commentHeader}\n\n${summary}\n\n${detailsHeader}\n${formattedStatsDiff}\n\n${recommendationsHeader}\n${recommendations}`;
+  }
+
+  /**
+   * Formats the difference in heap statistics into a readable string.
+   * @param {HeapDifferenceStatistics} statsDiff The difference in heap statistics.
+   * @returns {string} The formatted string.
+   */
+  private static formatHeapDifferenceStatistics(statsDiff: HeapDifferenceStatistics): string {
+    let message = '📊 **Memory Leak Detection Report** 📊\n\n';
+
+    statsDiff.forEach((entry) => {
+      message += `**GC Type**: ${entry.gcType}\n`;
+      message += `**Cost**: ${entry.cost.toLocaleString()} ms\n\n`;
+      message += '**Heap Statistics (before vs after executing the test)**:\n';
+      Object.entries(entry.diffGC.heapStatistics).forEach(([key, value]) => {
+        message += `- **${this.camelCaseToTitleCase(key)}**: ${this.formatBytes(value)}\n`;
+      });
+      message += '\n**Heap Space Statistics (before vs after executing the test)**:\n';
+      entry.diffGC.heapSpaceStatistics.forEach((space) => {
+        message += `  - **${this.snakeCaseToTitleCase(space.spaceName)}**:\n`;
+        Object.entries(space).forEach(([key, value]) => {
+          if (key !== 'spaceName') {
+            message += `    - **${this.camelCaseToTitleCase(key)}**: ${this.formatBytes(value)}\n`;
+          }
+        });
+        message += '\n';
+      });
+    });
+
+    return message;
+  }
+
+  /**
+   * Converts a string in camel case to title case.
+   * @param textInCamelCase The text in camel case.
+   * @return The text in title case.
+   */
+  private static camelCaseToTitleCase(textInCamelCase: string): string {
```

Comment:
This and snakeCaseToTitleCase() can potentially be used in a global scale so maybe consider switching it to public? Or well maybe we can address it in future PRs where these are used in global scope.

Reply:
Let’s address it in future PRs where it might be used, I barely got the CI to work for this PR 😅